x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/CrossClusterAccessServerTransportFilter.java

```diff
@@ -34,6 +46,39 @@ final class CrossClusterAccessServerTransportFilter extends ServerTransportFilte
         ALLOWED_TRANSPORT_HEADERS = Set.copyOf(allowedHeaders);
     }
 
+    // package private for testing
+    static final Set<String> CROSS_CLUSTER_ACCESS_ACTION_ALLOWLIST;
+    static {
+        final Stream<String> actions = Stream.of(
+            REMOTE_CLUSTER_HANDSHAKE_ACTION_NAME,
+            RemoteClusterNodesAction.NAME,
+            SearchAction.NAME,
```

Comment:
I've removed `TransportService.HANDSHAKE_ACTION_NAME` here -- an FC should never see this action, only `REMOTE_CLUSTER_HANDSHAKE_ACTION_NAME` on the new port.